src/core_tree_utils.c

```diff
@@ -129,7 +128,9 @@ int reorder_lhalo_to_lhvt(const int32_t nhalos, struct halo_data *forest, int32_
                     return EXIT_FAILURE;
                 }
             } else {
-                assert(desc >= 0 && desc < nhalos);
+                XASSERT(desc >= 0 && desc < nhalos, -1,
```

Comment:
This should be an `if` and then an `fprintf` + `return EXIT_FAILURE` (or appropriate error code) rather than an `XASSERT`.

(Or `XRETURN`)